qiskit/aqua/algorithms/minimum_eigen_solvers/qaoa/var_form.py

```diff
@@ -55,8 +55,13 @@ def __init__(self,
         self._num_qubits = cost_operator.num_qubits
         self._p = p
         self._initial_state = initial_state
-        self._num_parameters = 2 * p
-        self._bounds = [(0, np.pi)] * p + [(0, 2 * np.pi)] * p
+        if isinstance(mixer_operator, QuantumCircuit):
+            self._num_parameters = (1 + mixer_operator.num_parameters) * p
+            self._bounds = [(0, np.pi)] * p + [(0, 2 * np.pi)] * p * mixer_operator.num_parameters
```

Comment:
we should not add bounds to these parameters. Depending on the problem, the periodicity changes and can be much larger than 2*pi. I'd set bounds to `None`. This holds for both cases, using operator and circuit.

Reply:
Well, `2*pi` was in the initial code. I set to `None`.

Reply:
As discussed:

- cost operator --> [None, None]
- standard mixer --> [0, 2pi]
- custom mixer (doesn’t matter whether op or qc) --> [None, None]